src/process.rs

```diff
@@ -134,11 +137,147 @@
 ///            Expression::Paren(Box::new(Expression::Paren(Box::new(Expression::Letter('z'))))));
 /// # }
 /// ```
+///
+/// ### Sentence
+///
+/// To showcase the use of multiple matchers, we'll use a setence grammar:
+///
+/// ```no_run
+/// # #[macro_use] extern crate pest;
+/// # use pest::prelude::*;
+/// # fn main() {
+/// # impl_rdp! {
+/// # grammar! {
+/// sentence = { word ~ ([" "] ~ word)* }
+/// word     = { letter* }
+/// letter   = { ['a'..'z'] }
+/// # }
+/// # }
+/// # unreachable!();
+/// # }
+/// ```
+///
+/// A very simple AST that works in this case:
+///
+/// ```no_run
+/// # use std::collections::LinkedList;
+/// #[derive(Debug, PartialEq)]
+/// pub enum Node {
+///     Sentence(LinkedList<Node>), // we're using LinkedList because they're more efficient when
+///     Word(LinkedList<Node>),     // using tail recursion
+///     Letter(char)
+/// }
+/// ```
+///
+/// To build the `Token` processor, let's use a bottom up aproach when writing the matchers. Let's
+/// start by building a `word` matcher. We'll call it `_word` in order not to clash with the name
+/// of the rule that also gets defined as a method on the `Parser`.
+///
+/// ```ignore
+/// _word(&self) -> LinkedList<Node> { // return LinkedList<Node> to buil Word with
+///     (&head: letter, @tail) => { // usual tail recursion; &head is captured,
```

Comment:
so i'm not sure i understand what's happening there and the relation between `head` and `tail`. where do they end up?
At what point does `head` gets redefined? Is that a concept that `pest` knows about and handles automatically?